web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/ViewsBasicManager.php

```diff
@@ -16,14 +16,42 @@ class ViewsBasicManager extends ControllerBase implements ContainerInjectionInte
   /**
    * Allowed entity types for users to select.
    *
+   * Format:
+   * 'content_type_machine_name' => [
+   *   'label' => 'Human readable label',
+   *   'view_modes' => [
+   *     'view_mode_machine_name1' => 'Human readable label 1',
+   *     'view_mode_machine_name2' => 'Human readable label 2',
+   *   ],
+   * ],
+   *
    * @todo This seems fragile and would better be inside a config page for
    * admins to select.
    *
    * @var array
    */
+
   const ALLOWED_ENTITIES = [
-    'node' => 'node_type',
-    // 'media' => 'media_type',
+    'news' => [
+      'label' => 'News articles',
+      'view_modes' => [
+        'card' => 'News cards',
+        'list_item' => 'News list items',
+      ],
+    ],
+    'event' => [
+      'label' => 'Events',
+      'view_modes' => [
+        'card' => 'Event cards',
+        'list_item' => 'Event list items',
+      ],
+    ],
+    'page' => [
+      'label' => 'Pages',
+      'view_modes' => [
+        'teaser' => 'Teasers',
+      ],
+    ],
```

Comment:
Like you, I would love to replace this with something Drupalier. Let's keep on this path for a little to see how ugly things get. This current path gives us a lot of freedom and flexibility. We can create something more structured once we get some more bells and whistles.